### PR TITLE
Puzzle scene

### PR DESCRIPTION
**Pull request by Ittis**
Have fun

### CODE REVIEW

File: Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as [hunk 1]
```diff
@@ -12,8 +12,8 @@ class Minecart_Puzzle_1{
     Minecart_Puzzle_1(Entity @entity){
         @hub = Managers();
         @self = @entity;
-        @selfBody = self.GetRigidBody();
-        @puzzleBody = self.GetParent().GetParent().GetChild("BreakCheck").GetRigidBody();
+        //@selfBody = self.GetRigidBody();
```
Comment (NinjaDanz3r) [inline Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: Remove commented lines.

Comment (Ittis) [reply Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: Fixed

File: src/Engine/Manager/ScriptManager.cpp [hunk 2]
```diff
@@ -65,15 +65,18 @@ void AngelScriptMessageCallback(const asSMessageInfo* message, void* param) {
 void AngelScriptDebugLineCallback(asIScriptContext *ctx, const std::map<std::string, std::set<int>>* breakpoints){
     const char *scriptSection;
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MINOR Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L583](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L583): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L847](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L847): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L853](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L853): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L861](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L861): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/ScriptManager.cpp [hunk 3]
```diff
@@ -65,15 +65,18 @@ void AngelScriptMessageCallback(const asSMessageInfo* message, void* param) {
...
     int line = ctx->GetLineNumber(0, 0, &scriptSection);
+    asIScriptFunction *function = ctx->GetFunction();
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MINOR Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L585](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L585): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L849](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L849): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L855](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L855): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L863](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/407f11869997f2e6ccc30740ae0d82f4b1ddea50/src/Engine/Manager/ScriptManager.cpp#L863): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**